### ISSUE TITLE
Modification of gradients

### ISSUE THREAD

**Issue by deasmhumnha** [gorgonia/gorgonia#245]
Can you currently modifying gradients without changing forward computations? I see there is the `SetDerivOf` method; would it be as simple as creating a `GetDerivOf` method? Use cases are any number of specialized fitting algorithms (gradient clipping, gradient normalization, projection, etc.). My particular application is performing variance reduction on stochastic gradients.

**Comment by deasmhumnha** [comment]
True. I imagine this would work best as a function called during backpropagation in order for the unmodified gradients to still be accessible during each passing through the graph.
